assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * SurveyQuestionMultiSelect component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import keyBy from 'lodash/keyBy';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../Button';
+import Checkbox from '../Checkbox';
+import SurveyHeader from './SurveyHeader';
+import { TextField, Input } from '../../material-components';
+
+const MAXIMUM_CHARACTER_LIMIT = 100;
+
+// Why have this rule enabled if we get this from the APIs *and* send back to the APIs like this?
+/* eslint-disable camelcase */
+
+const SurveyQuestionMultiSelect = ( { question, choices, answerQuestion, dismissSurvey, min_choices = 1, max_choices } ) => {
+	const mappedChoices = choices.map( ( { answer_ordinal, write_in } ) => {
+		const optionalKeys = write_in ? { answer_text: ''	} : {};
+
+		return {
+			answer_ordinal,
+			selected: false,
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
```

Comment:
There is a lot of filtering and reducing here.  Which makes me think that
1. `useReducer` may be better (or an off-the-shelf hook from a popular React hooks library)
2. Some react-testing-library tests would be good here

Reply:
Now there are comprehensive tests I can refactor this pretty easily! :+1: